packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -613,12 +607,6 @@ void _validateFlutter(YamlMap? yaml, List<String> errors) {
         break;
       case 'deferred-components':
         _validateDeferredComponents(kvp, errors);
-      case 'disable-swift-package-manager':
-        if (yamlValue is! bool) {
-          errors.add(
-            'Expected "$yamlKey" to be a bool, but got $yamlValue (${yamlValue.runtimeType}).',
-          );
-        }
```

Comment:
If people still have the old setup in their config, won't it throw an error (see line 617 below)?

Perhaps instead of removing completely, throw a guided error here explaining how to fix

Reply:
Done!